src/re2parser.cc

```diff
@@ -202,7 +202,7 @@ namespace {
                                 symbols.push_back(symbol);
                                 // Foldcase causes RE2 to do a case-insensitive match, so transitions will be made for
                                 // both uppercase and lowercase symbols
-                                if (inst->foldcase()) {
+                                if (inst->foldcase() && symbol >= 'a' && symbol <= 'z') {
                                     symbols.push_back(symbol-ascii_shift_value);
                                 }
                             }
```

Comment:
Shouldn't there also be something like
```
if (inst->foldcase() && symbol >= 'A' && symbol <= 'Z') {
                                    symbols.push_back(symbol+ascii_shift_value);
                                }
                            }
```
So we also add lowercase if we have uppercase. Or does RE2 automatically work only with lowercase if case-insensitive matching is enabled?

Reply:
This condition is unnecessary because RE2 performs `foldcase` by mapping uppercase to lowercase. As a result, this condition would never be used.

Reply:
You are correct. The value depends on RE2's decision on whether to use foldcase mode or not. The comment https://github.com/VeriFIT/mata/pull/463#issuecomment-2483729849 elaborates in detail on how RE2 makes this decision.